go/pkg/filemetadata/filemetadata.go

```diff
@@ -41,11 +47,25 @@ func isSymlink(filename string) (bool, error) {
 func Compute(filename string) *Metadata {
 	md := &Metadata{Digest: digest.Empty}
 	file, err := os.Stat(filename)
+	if isSym, _ := isSymlink(filename); isSym {
+		md.Symlink = &SymlinkMetadata{}
+		if err != nil {
+			md.Err = &FileError{Err: err}
+			md.Symlink.IsDangling = true
+			return md
+		}
+		dest, err := os.Readlink(filename)
```

Comment:
Note that I used `os.Readlink` instead of [`filepath.EvalSymlinks()`](https://golang.org/pkg/path/filepath/#EvalSymlinks). From what I can tell, the later evaluates to its physical path, .e.g. `/tmp/foo -> bar` becomes `/tmp/foo -> /private/tmp/bar` on Mac. My reasoning is that it is up to the upper-level code to decide how to re-construct the symlink path, so just returning the target is more appropriate, WDYT?

Reply:
Agreed! We can revisit this later if we find out that we need something else in the caller code.